src/conn_state.c

```diff
@@ -60,7 +60,9 @@ on_poll(uv_poll_t* handle, int status, int events)
         return;
 
     enftun_netlink_read_message(&conn_state->nl, msg_buf, sizeof(msg_buf));
-    check_preferred_route(conn_state);
+    int rc = check_preferred_route(conn_state);
```

Comment:
Also update `check_preferred_route` to return 0 on success.  Currently it returns non-zero if the addresses aren't equal and it calls `reconnect_cb` itself.